### ISSUE TITLE
refit_full=True

### ISSUE THREAD

**Issue by yaniv92648** [autogluon/autogluon#5071]
### Describe the issue linked to the documentation

Hi,

In the [TimeSeriesPredictor](https://auto.gluon.ai/stable/api/autogluon.timeseries.TimeSeriesPredictor.html) constructor, I'm passing prediction_length=90 & freq='D' (to predict for 90 days).
In the [TimeSeriesPredictor.fit](https://auto.gluon.ai/stable/api/autogluon.timeseries.TimeSeriesPredictor.fit.html) method, I'm passing num_val_windows=1 & refit_full=True.
So, TimeSeriesPredictor trains according to the training data of the historical data up to 90 days ago, and then validates the best model according to the validation set of the recent 90 days.
Because I'm passing refit_full=True, so I think that after finding the best model according to the validation set, TimeSeriesPredictor trains this best model over all of the recent data (including the 90 days that were used for the validation set earlier, to find the best model in the 1st place), and using this best model to predict  future data with.

Is the above correct?

I'm asking because as we can see in the attached [documentation](https://auto.gluon.ai/stable/api/autogluon.timeseries.TimeSeriesPredictor.fit.html), in the refit_full parameter, it says "AutoGluon will attempt to re-train all models" and not "AutoGluon will attempt to re-train the best model".
![refit](https://github.com/user-attachments/assets/af62ced1-58f5-4c27-93d8-c3bc0b724ced)

Thanks!

**Comment by shchur** [comment]
Hi @yaniv92648, thank you for the question! If you set `predictor.fit(..., refit_full=True)`, the predictor will attempt to refit all models (that support refitting without validation data) using all available data. This is equivalent to calling
```python
predictor.fit(...)
predictor.refit_full()  # defaults to predictor.refit_full("all")
```

If you want to only refit the best model, you will need to change the above code to
```python
predictor.fit(...)
predictor.refit_full("best")
```
If the best model is the ensemble, the ensemble itself won't be retrained, but all of its constituent models will be retrained (you will see a `_FULL` suffix).

One important caveat is that not all forecasting model support refitting:
- Local models like ETS and pretrained models like Chronos are always automatically re-fitted during `predict`, so you can think of these models as always performing "refit_full".
- DL models from GluonTS like DeepAR and PatchTST require a validation set during training, so they cannot be refitted.
- Tabular models can actually be retrained on full data.
